__tests__/decorator.ts

```diff
@@ -0,0 +1,62 @@
+import { EMPTY, Subject } from 'rxjs';
+
+import * as untilDestroyedObj from '../src/take-until-destroy';
+
+import { WithUntilDestroyed } from '../src/decorator';
+
+describe('@WithUntilDestroyed decorator', () => {
+  it('should throw when applied on non Observable prop', () => {
+    class Test {
+      @WithUntilDestroyed()
+      notObservable = true;
+    }
+
+    expect(() => new Test()).toThrowError();
+  });
+
+  it('should call `untilDestroyed()` with target prototype and `destroyMethodName` on setter', () => {
+    const untilDestroyedSpy = spyOn(
+      untilDestroyedObj,
+      'untilDestroyed',
+    ).and.callThrough();
+
+    class Test {
+      @WithUntilDestroyed('destroy')
+      notObservable = EMPTY;
```

Comment:
what do u mean "notObservable"? `EMPTY ` is observable.

Reply:
Yes, it was a copy/paste from other test where is made sense. Will update